blazar/db/sqlalchemy/utils.py

```diff
@@ -163,16 +165,51 @@ def get_reservation_allocations_by_host_ids(host_ids, start_date, end_date,
     session = get_session()
     border0 = models.Lease.end_date < start_date
     border1 = models.Lease.start_date > end_date
-    query = (session.query(models.Reservation, models.ComputeHostAllocation)
-             .join(models.Lease).join(models.ComputeHostAllocation)
-             .filter(models.ComputeHostAllocation.compute_host_id
-                     .in_(host_ids))
-             .filter(~sa.or_(border0, border1)))
+    fields = ['id', 'status', 'lease_id', 'start_date',
+              'end_date', 'lease_name', 'project_id']
+
+    reservations_query = (session.query(
+        models.Reservation.id,
```

Comment:
I wonder if _this_ is the real optimization: pulling specific columns rather than the entire thing.

Reply:
I don't think I broke down the columns in the original query because breaking it up had such a big effect. The additional filters took the result down from a second to half a second. Breaking down the columns on the smaller queries didn't really have an effect, but maybe it would on the original query.